### ISSUE TITLE
Performance issue with layout reflows. Implementation question

### ISSUE THREAD

**Issue by andresgutgon** [FezVrasta/popper.js#54]
### What?
First of all I know this lib is only managing tooltip positioning. Not how user open/close or toggle it.
What I'm trying is to wrap a PopperJS instance inside a boostrap tooltip like jQuery plugin. This way I can handle user clicks and open/close tooltips.

## What I'm doing
I'm creating PopperJS instance the first time user clicks on the reference.

``` javascript

show: function() {
  ....
  if (!this.popper_instance) {
      this.popper_instance = new PopperJS(
        this.$element,
        popperObject,
        {
          ...POPPER_DEFAULT_OPTIONS,
          ...popperOptions,
          modifiers: this.getModifiers(),
        },
      );
      
      this.$tip = $(this.popper_instance._popper);
  }

  this.$tip
    .detach()
    .appendTo('body');
  ...
}
```

But initiating a Popper instance requires get `offsetHeight` witch makes a lot of layouts reflows.

## Insert in the body
Something important that I'm doing is inserting the tooltip in `body` page. Is that ok?

Does somebody a good example of how connect popper into a jquery plugin?

## DOM size
Another thing is that my page has a lot DOM nodes

**Comment by FezVrasta** [comment]
Please provide a codepen to reproduce the problem.

**Comment by andresgutgon** [comment]
It's not a problem. Popper is working right. It's an implementation question.

## My questions: 

**When is the best moment to create an instance of Popper?**
I'm doing when user clicks on the button. I can't do when I introduce my tooltips because content is dynamic and I don't want to load the content of all my poppers when page loads.

**Is a good idea insert popper inside body as a direct child?**
This is good because it allows me control the `z-index` order and ensure my tooltips are always on top. But this is also causing a lot of [Browser Layout repaints](http://kellegous.com/j/2013/01/26/layout-performance/) (in a page with a lot of DOM nodes)

As I said is not a bug. But it would be great if we can explain a list of best practice on how to integrate it within an app. Sorry if you don't considere this is interesting. Just ignore my comment and move on :)

Thanks for the lib. It's great. Thanks for your time.

**Comment by FezVrasta** [comment]
yes that's why I'm asking for a codepen demo, so I can take a look at it and give u advices

**Comment by andresgutgon** [comment]
Hi, sorry for the late response:
http://codepen.io/andresgutgon/pen/OXjKoO

In the codepen. I'm using `jquery`, `underscore` and `popper` as dependencies. And also the `tooltip.js` that I'm using.

This [tooltip.js](https://gist.github.com/andresgutgon/3e19c3b7f25d003125a2c605f3f04d03) is a copy of `boostrap-tooltip-2.3.2.js` but addapted to use `Popper` as positioning engine

## Detach or not detach
My initial approach was `detach` the tooltip content and `attach` on `show`. But this was causing a lot of reflows. You can play with the option and run a performance test with Chrome Dev Tools
